File: src/front/models/track.ts
import { Artist } from "@models/artist";

export type TrackStatus = "ok" | "unavailable" ;

export class Track {
    public idTrack?: number;
    public title?: string;
    public youtubeLink?: string;
    public listeningCount?: number;
    public durationSeconds?: number;
    public createdAt?: string;
    public artists?: Artist[];
    public status?: TrackStatus;

    constructor(data: Partial<Track>) {
        Object.assign(this, data);
    }

    isPlayable(): boolean {
        return this.status === "ok" || this.status === undefined;
    }
}

File: src/cli/src/models/track.py
from pydantic import BaseModel
from typing import List, Literal
from datetime import datetime
from .artist import ArtistSchema

TrackStatus = Literal["ok", "unavailable"]


class TrackSchema(BaseModel):
    idTrack: int
    title: str
    youtubeLink: str | None = None
    listeningCount: int
    durationSeconds: int | None = None
    createdAt: datetime
    artists: List[ArtistSchema]
    status: TrackStatus = "ok"

    def is_playable(self) -> bool:
        return self.status == "ok"
